### DISCUSSION TITLE
Is there way to parse a Cypress configuration file and obtain its values?

### DISCUSSION THREAD

**Discussion by mbolotov** [cypress-io/cypress#24866] [Questions and Help]
Cypress Support plugins for IntelliJ (both [Base ](https://plugins.jetbrains.com/plugin/13819-cypress-support)and [Pro](https://plugins.jetbrains.com/plugin/13987-cypress-support-pro)) require the data from the cypress configuration in order to make correct suggestions in the code and so on. 
It was easy to parse the configuration file in JSON format prior to Cypress 10 version. 
Is there a way to do this in Cypress 10 and above?
I think it would be nice to have some official API for doing that. Beside the plugins, there may be other uses for this API as well.

**Comment by portyu9** [comment]
Hi @mbolotov 

There still is not a documented public Cypress API whose job is simply:

```text
config file -> fully resolved Cypress configuration
```

without starting Cypress.

This is fundamentally different from the old `cypress.json` case because `cypress.config.js`, `.ts`, `.mjs`, and `.cjs` are **executable modules**, not just data files. The configuration can depend on Node code, environment variables, asynchronous work, and changes made by `setupNodeEvents()`.

If you control the Cypress project, the supported place to access the resolved configuration is `setupNodeEvents()`:

```js
const { defineConfig } = require('cypress')

module.exports = defineConfig({
  e2e: {
    setupNodeEvents(on, config) {
      console.log(config)
      return config
    },
  },
})
```

That `config` object contains Cypress's resolved configuration at that stage, plus values such as `configFile` and `projectRoot`.

The Node Module API also accepts `configFile`:

```js
const cypress = require('cypress')

cypress.run({
  configFile: 'cypress.config.js',
}).then((results) => {
  console.log(results.config)
})
```

but of course that actually starts a Cypress run, so it is not a lightweight parser API for an IDE.

For an IntelliJ plugin I would **not** try to statically parse arbitrary JS/TS configuration and treat the result as authoritative. You could handle simple literal configs that way, but it cannot reproduce Cypress's resolved configuration in the general case because the config is executable.

So an official "resolve config only" API would still be useful for exactly the tooling scenario you described.

Please mark this answer as accepted if it helped, thank you.
